test/gtest/ucp/test_ucp_tag_match.cc

```diff
@@ -369,4 +369,141 @@ UCS_TEST_P(test_ucp_tag_match, sync_send_unexp) {
     request_release(my_send_req);
 }
 
+UCS_TEST_P(test_ucp_tag_match, sync_send_unexp_rndv, "RNDV_THRESH=1048576") {
+    static const size_t size = 1148576;
+    request *my_send_req;
+    ucp_tag_recv_info_t info;
+    ucs_status_t status;
+
+    std::vector<char> sendbuf(size, 0);
+    std::vector<char> recvbuf(size, 0);
+
+    ucs::fill_random(sendbuf.begin(), sendbuf.end());
+
+    /* sender - send the rts*/
+    my_send_req = send_sync_nb(&sendbuf[0], sendbuf.size(), DATATYPE, 0x111337);
+    /* receiver - get the rts and put in unexpected */
+    short_progress_loop();
+
+    ASSERT_TRUE(my_send_req != NULL);
+    EXPECT_FALSE(my_send_req->completed);
+
+    /* receiver - issue a recv req, match the rts, perform rndv-get and send ats to sender */
+    status = recv_b(&recvbuf[0], recvbuf.size(), DATATYPE, 0x1337, 0xffff, &info);
+    ASSERT_UCS_OK(status);
+
+    EXPECT_EQ(sendbuf.size(), info.length);
+    EXPECT_EQ((ucp_tag_t)0x111337, info.sender_tag);
+    EXPECT_EQ(sendbuf, recvbuf);
+
+    /* sender - get the ATS and set send request to completed */
+    ucp_worker_progress(sender->worker());
+
+    EXPECT_TRUE(my_send_req->completed);
+    EXPECT_EQ(UCS_OK, my_send_req->status);
+    request_release(my_send_req);
+}
+
+UCS_TEST_P(test_ucp_tag_match, rndv_req_exp, "RNDV_THRESH=1048576") {
+    static const size_t size = 1148576;
+    request *my_send_req, *my_recv_req;
+
+    std::vector<char> sendbuf(size, 0);
+    std::vector<char> recvbuf(size, 0);
+
+    ucs::fill_random(sendbuf.begin(), sendbuf.end());
+
+    /* sender - send the RTS */
+    my_send_req = send_nb(&sendbuf[0], sendbuf.size(), DATATYPE, 0x111337);
```

Comment:
this test is called 'exp' but i see send is done before receive, so you the assuumption is if we don't do progress it will not arrive as unexpected?
maybe better to call recv_nb first and then send_b to make sure this is expected?
did you copy it from a previous test (which means it should be fixed too)?